File: lib/src/main/java/com/sd/lib/umeng/push/LibUmengPush.java
package com.sd.lib.umeng.push;

import android.content.Context;

import com.umeng.message.PushAgent;
import com.umeng.message.api.UPushRegisterCallback;

public class LibUmengPush {
    private LibUmengPush() {
    }

    /**
     * 注册
     */
    public static void register(Context context, UPushRegisterCallback callback) {
        PushAgent.getInstance(context).register(callback);
    }
}

File: lib/src/main/java/com/sd/lib/umeng_push/LibUmengPush.kt
package com.sd.lib.umeng_push

import android.content.Context
import com.umeng.commonsdk.utils.UMUtils
import com.umeng.message.PushAgent
import com.umeng.message.api.UPushRegisterCallback

object LibUmengPush {
    @JvmStatic
    fun register(context: Context, callback: UPushRegisterCallback?) {
        if (UMUtils.isMainProgress(context)) {
            PushAgent.getInstance(context).register(callback)
        }
    }
}